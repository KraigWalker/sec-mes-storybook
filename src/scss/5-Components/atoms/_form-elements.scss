$form-border-color: bow-palette(grey, x-dark);
$form-border-width: 1px;
$form-border-radius: 5px;
$form-disabled-color: bow-palette(grey, disabled-text);
$form-disabled-bg-color: bow-palette(grey, disabled-cell);
$form-label-color: bow-palette(black, base);
$text-color: bow-palette(grey, text);

%focus {
	@include on-focus();
}

.c-field,
.c-preview {
	@include clearfix;
	margin-top: 0;
	margin-bottom: 24px;
	font-family: 'Muli', sans-serif;

	.c-field__extra {
		font-size: 14px;
		display: block;
		font-weight: normal;
	}

	/*
		LABELS
	 */
	.c-field__label,
	.c-preview__label {
		@include clearfix;
		@extend %type--arran;

		display: block;
		width: 100%;
		color: $form-label-color;
		margin-bottom: 16px;
		border: none;
        font-weight: bold;
        position: relative;

		@include media-breakpoint-up(sm1) {
			vertical-align: middle;
			display: block;
			float: left;
		}

		small {
			color: #999;
		}
	}

	.c-field__label--block {
		display: block;
		clear: left;
	}

	.c-field__instructions {
		@extend %type--jura;
	}

/*
	Custom dropdown using button and list
*/
	.c-dropdown {
		text-align: left;

		&__list {
			background-color: bow-palette(white, base);
			position: absolute;
			border: 1px solid;
			border-radius: 5px;
			list-style: none;
			padding : 0px;
			z-index: 2;
			width: 100%;
		}

		&__value {
			padding: 3px 12px;

			&:first-child {
				border-top-left-radius: 5px;
				border-top-right-radius: 5px;
			}

			&:last-child {
				border-bottom-left-radius: 5px;
				border-bottom-right-radius: 5px;
			}
	
			&:hover {
				background-color: bow-palette(gray,light);
			}
		}
	}

	.char__error {
		color: red;
	}

	.c-field__controls,
	.c-field__message,
	.c-preview__controls,
	.c-preview__message {
		@include clearfix;
		@extend %type--jura;
		margin-left: 0;
		margin-bottom: 5px;
		input,
		select,
		textarea,
		.c-dropdown,
		.c-dropdown__list,
		.input-group.date {
			max-width: 343px;
			font-size: inherit;
			color: bow-palette(grey, text);
			@include media-breakpoint-up(md1) {
			    max-width: 304px;
			}

			@include media-breakpoint-up(lg1) {
			    max-width: 374px;
			}

			@include media-breakpoint-up(xl1) {
			    max-width: 428px;
			}

			&[type=text]{
				background-color: bow-palette(white, base);
			}
		}

		label {
			input {
				max-width: none;
			}
		}
	}

	.c-field__controls,
	.c-preview__controls {
		display: block;
		clear: both;
		.callout.callout__error {
			margin-top: 5px;
			margin-bottom: 0;
			max-width: 343px;
			@include media-breakpoint-up(md1) {
			    max-width: 304px;
			}

			@include media-breakpoint-up(lg1) {
			    max-width: 374px;
			}

			@include media-breakpoint-up(xl1) {
			    max-width: 428px;
			}
		}
	}

	.c-field__controls label {
		line-height: 20px;
		vertical-align: middle;
	}


	.c-field__input,
	.c-field__input[type=text],
	.c-preview__input {
		@include brand-text ();
		height: 48px;
		margin: 0;
		padding: 2px 16px;
		box-shadow: none;
		width: 100%;
		border: $form-border-width solid bow-palette(grey, x-dark);
		border-radius: $border-radius;
		min-height: 48px;
		color: bow-palette(grey, text);
		background-color: bow-palette(white,base);


        &:focus {
			@include on-focus();
			border: 1px solid bow-palette(grey, x-dark);
			position: relative;
			z-index: 1;	
        }
	}

	.c-field__input--txtarea {
		min-height: 364px;
		padding: get-px(b);
		resize: none;
	}

	.c-field__input--select {
		@include drop-brand-color();
		-webkit-appearance: none;
		-moz-appearance: none;
		padding-right: 48px;
	}
	
	.c-preview__input {
		display: block;
		float: left;
		border: 2px solid #ccc;
		background-color: #efefef;
		color: #666;
		line-height: 40px;
		padding: 0 10px;
	}

	.c-field__option,
    .c-field__extra {
        color: #7b7b7b;
    }

    .c-field__instructions {
        font-weight: normal;
        color: #777;
    }

	.c-field__message,
	.c-preview__message {
		@extend %type--jura;
		display: block;
        width: 100%;

        @media (max-width: 480px) {
        	margin-left: 0;
        }

    }

	.c-field__input--small[type=text] {
		width: 200px;
	}

	.c-field__input[type=checkbox] + span {
        padding-left: 0;
        padding-right: 0;
        margin-left: 0;
		margin-right: 0;
		padding: 0;
		cursor: pointer;
		display: block;

        &:before {
			
            border-radius: 5px;
			border-width: 1px;
			border-color: #c9c9c9;
			background-color: bow-palette(white, base);
			height: 24px;
			width: 24px;
			display: inline-block;
			left: 0;
			border: 1px solid palette((color: black)...);
			content: ' ';
			position: relative;
			margin-right: 16px;
			vertical-align: middle;
			top: 1px;
        }
	}

	.c-field__input[type=checkbox]:checked + span::before {
		background: none;
		content: "\48";
		font-family: "webfonts";
		font-size: 27px;
    	line-height: 22px;
    	text-indent: -2px;
	}
	

    .c-field__input[type=checkbox]:focus + span {
        outline: 0;

        &:before {
			@include on-focus();
        }
    }
        

    .c-field__input[disabled] {
        background-color: $form-disabled-bg-color;
		border-color: $form-disabled-color;
    }
	
    .c-field__message {
        margin-top: 10px;
		display: none;
    }

	.c-field__controls {
		.c-field__message {
			* {
				font-family: 'Muli', sans-serif;
			}
			p {
				margin: 0;
			}
		}
	}
}

.c-field--sortcode {
	.c-field__controls {
		.c-field__input,
		.c-field__submit {
			display: block;
			float: left;

		}
		.c-field__input[type=text] {
			width: 40px;
			margin-right: 10px;
		}
		.c-field__submit--inline {
			margin-top: 0;
			margin: 0;
		}
	}
}

.c-field--error {
	.c-field__label,
   	.c-field__message {
		display:block;
   	}
	.c-field__controls {
		.c-field__input {
			border-color: bow-palette(red, error);
		}
		.c-field__message{
			padding: 10px;
			border-radius: 8px;
			background: bow-palette(red,x-light);
			border: 1px solid;
			border-color: bow-palette(red,dark);
		}
	}
}



.c-field {

	.c-field__controls {


		/***********************************************\
	      RADIOS + CHECKBOXES STYLES
	    \***********************************************/

		// Reset for browser radios + checks
		.c-field__input[type="checkbox"],
		.c-field__input[type="radio"]{
			display: inline;
			position: absolute;
			margin: 0;
			height: 0;
			width: 0;
			opacity:0;

			&:focus{
				outline: none;
			    outline-width: 0;

			}

		}

		// This is the label which must come after
		.c-field__input[type="checkbox"] 	+ .c-field__label,
		.c-field__input[type="radio"] 	+ .c-field__label {
			padding-left: 30px;
			padding-right: 30px;
			position: relative;

		}

		// Fake radio and check is the :before element of the label
		.c-field__input[type="checkbox"] 	+ .c-field__label:before,
		.c-field__input[type="radio"] 	+ .c-field__label:before {
			content: "";
			width: 20px;
			height: 20px;
			border: 2px solid #c9c9c9;
			top: 14px;
			left: 5px;
			@include common-theme(background, white, base);
			position: absolute;
			//box-shadow: inset 0 2px 0 0 #d5d5d5;
			-webkit-transition: all .2s;
				-moz-transition: all .2s;
				transition: all .2s;

		}

		// Checked State
		.c-field__input:checked + .c-field__label:before {
			border-color: #2e3137;
			box-shadow: inset 0 0 0 0 bow-palette(white, base);

		}

		// Disabled State
		.c-field__input:disabled + .c-field__label {
			color: $form-disabled-color;

			&:before,
			&:active:before{
				border-color: $form-disabled-color;
				box-shadow: none;

			}

		}

		// Focus State
		.c-field__input:focus 				+ .c-field__label:before,
		.c-field__input[data-error]:focus  	+ .c-field__label:before {
			box-shadow: 0 0 0 2px bow-palette(orange, base);
		}


		/***********************************************\
	      RADIOS SPECIFIC STYLES
	    \***********************************************/

		.c-field__input[type="radio"]{

			// :before is the fake radio
			& + .c-field__label:before{
				border-radius: 50%;

			}

			// :after is the dot
			// Dot closed
			& + .c-field__label:after{
				-webkit-transition: all .2s;
				-moz-transition: all .2s;
				transition: all .2s;
				content: "";
				width: 0;
				height: 0;
				left: 15px;
				top: 24px;
				background: #2e3137;
				position: absolute;
				border-radius: 50%;
				@include rotate(45deg);

			}

			// Dot open
			&:checked  + .c-field__label:after{
				width: 10px;
				height: 10px;
				left: 10px;
				top: 19px;
				@include rotate(0deg);

			}

		}
		&.csstransitions {
			.c-field__input[type="radio"] 	+ .c-field__label:after{
				height: 10px;
				width: 10px;
				-webkit-transform: scale(0);
				-webkit-transform-origin: 0% 0%;
				-moz-transform: scale(0);
				-moz-transform-origin: 0% 0%;
				transform: scale(0);
				transform-origin: 0% 0%;
			}
			.c-field__input[type="radio"]:checked 	+ .c-field__label:after{
				top: 5px;
				left: 5px;
				-webkit-transform: scale(1);
				-moz-transform: scale(1);
				transform: scale(1);
			}
		}

		/***********************************************\
	      CHECKBOX SPECIFIC STYLES
	    \***********************************************/

		.c-field__input[type="checkbox"]{

			& + .c-field__label:after {
				@include rotate(-45deg);
				-webkit-transition: all .2s;
				-moz-transition: all .2s;
				transition: all .2s;
				height: 20px;
				width: 20px;
				position:absolute;
				top: 0;
				left: 10px;
				//font-family: 'testtttt';
				speak: none;
				font-style: normal;
				font-weight: normal;
				font-variant: normal;
				text-transform: none;
				line-height: 20px;
				font-size: 0;
				-webkit-font-smoothing: antialiased;
				content: "\ea10";
			}

			&:checked + .c-field__label:after,
			.c-field__label--checked:after{
				@include rotate(0deg);

				-webkit-transition: all .2s;
				-moz-transition: all .2s;
				transition: all .2s;
				height: 20px;
				width: 20px;
				left: 4px;
			}
		}
	}

	/*
		PREPEND + APPEND
	 */

	.c-field__prepend,
	.c-field__append {
		display: flex;
		width: 100%;
		

		.c-field__addon {	
			line-height: 40px;
			height: 48px;
			width: 48px;
			text-align: center;
			background-color: bow-palette(white, base);
			// @include common-color-base-white(color, white, base);
			border-top: $form-border-width solid $form-border-color;
			border-bottom: $form-border-width solid $form-border-color;
			min-height: 48px;
			padding: 2px 8px;
			
		}	

		.c-field__action,
		.c-field__icon {
			@extend %type--jura;
			border: $form-border-width solid $form-border-color;
			@include general-theme(background-color, blue, account);
			@include common-color-base-white(color, white, base);
			border-left: 0;
			border-radius: 5px;
			border-top-left-radius: 0;
			border-bottom-left-radius: 0;
			
			
			height: 48px;
			
			padding-top: 2px;
			padding-bottom: 2px;

		}

		.c-field__icon {
			font-size: 24px;
			line-height: 48px;
		}
	}

	.c-field__prepend {
		.c-field__addon {
			border-top-left-radius: 5px;
			border-bottom-left-radius: 5px; 
			border-left: $form-border-width solid $form-border-color;
		}

		.c-field__addon + .c-field__input {
			border-top-left-radius: 0;
			border-bottom-left-radius: 0;
		}
	}

	.c-field__append {
		.c-field__addon {
			border-top-right-radius: 5px;
			border-bottom-right-radius: 5px; 
			border-right: $form-border-width solid $form-border-color;
		}

		.c-field__input {
			border-top-right-radius: 0;
			border-bottom-right-radius: 0;
		}

		.c-field__input + .c-field__addon {
			border-top-left-radius: 0;
			border-bottom-left-radius: 0;
		}
	}

}

.c-field--row,
.c-preview--row {
	.c-field__label,
	.c-preview__label {
		line-height: 48px;
		padding-top: 0;
	}

	.c-preview__controls {
		.c-preview__input {
			padding-left: 16px;
			padding-right: 16px;
			background-color: #f7f7f7;
			border: 0;
			line-height: 48px;
		}
	}
}

.c-field--radios {
	.c-field__controls {
		display: flex;
		flex-wrap: wrap;

		.c-field__radio {
			display: block;
			float: left;
			width: 100%;
			vertical-align: top;

			&:hover:after {
					background-color: red;
				}

			.c-field__label {
				border: 1px solid transparent;
				border-radius: 100px;
			}
		}

		.c-field__label {
			padding: 10px;
			margin: 0;
			min-height: 48px;
			padding-top: 14px;
			padding-bottom: 14px;
		}

		.c-field__radio--inline {
			display: inline-block;
			float: none;
			width: auto;
		}
	}

	
}

.c-field--radios--with-border {
	.c-field__controls {

		.c-field__label {
			font-weight: normal;
		}

		.c-field__input[type=radio] + .c-field__label:hover {
			&:after {
				background-color: $text-color;
				height: 10px;
				width: 10px;
				left: 10px;
				content: "";
				top: 19px;
				transition: none;
			}
		}

		.c-field__radio {
			margin-bottom: 5px;
			border-radius: 100px; // per Forms_v1_07/07.17

			.c-field__label {
				border: 2px solid $text-color;
			}

			.c-field__input + .c-field__label {
				&:before {
					border-color: $text-color;
				}

			}

			.c-field__input:checked + .c-field__label {
				background-color: $text-color;
				border-color: $text-color;
				border-radius: 100px;
				@include common-theme(color, white, base);

				&:before {
					@include common-theme(border-color, white, base);
					background-color: $text-color;
					border-width: 2px;
					border-style: solid;
				}

				&:after {
					@include common-theme(background-color, white, base);
				}
			}

			.c-field__input:disabled + .c-field__label {
				border-width: 1px;
				border-color: $form-disabled-color;
				color: $form-disabled-color;
				
				&:before {
					border-color: $form-disabled-color;
				}
			}

			
		}
	}
}

.c-field--checkbox {
    
    .c-field__label {
        position: relative;
        
        input[type=checkbox] {
            position: absolute;
            top: 0;
			left: 0;
			opacity: 0;
        }
    }
}

.c-field--disabled {
	.c-field__label {
		color: $form-disabled-color;
	}

	.c-field__controls {
		.c-field__input[type=text] {
			background-color: bow-palette(grey, disabled-cell, base);
			color: bow-palette(grey, disabled-text, base);
			-webkit-text-fill-color: bow-palette(grey, disabled-text, base);
			opacity: 1;
		}
	}
}