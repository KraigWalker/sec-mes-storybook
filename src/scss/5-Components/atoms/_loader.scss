// Added new spinner loader
@keyframes rotating {
    from {
        -ms-transform: rotate(0deg);
        -moz-transform: rotate(0deg);
        -webkit-transform: rotate(0deg);
        -o-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    to {
        -ms-transform: rotate(360deg);
        -moz-transform: rotate(360deg);
        -webkit-transform: rotate(360deg);
        -o-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

.spinner-loader {
    display: table;
    margin: 0 auto;
    -webkit-animation: rotating 1s linear infinite;
    -moz-animation: rotating 1s linear infinite;
    -ms-animation: rotating 1s linear infinite;
    -o-animation: rotating 1s linear infinite;
    animation: rotating 1s linear infinite;
}

.page-loader {
    position: fixed;
    top: 35%;
    left: 50%;
}