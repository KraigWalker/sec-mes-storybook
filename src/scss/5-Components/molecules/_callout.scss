$calloutwidths: ( xs1 : 343px, xs2 : 343px, sm1 : 720px, sm2 : 720px, md1 : 723px, lg1 : 1022px, lg2 : 1022px, lg3 : 1022px, xl1 : 1380px);
@mixin calloutsize($map, $key) {
    max-width: map-get($calloutwidths, $key);
}

.callout {
    padding: 16px;
    @include common-theme(background-color, blue, x-light);
    border-radius: 5px;
    margin-bottom: 20px;
    border: 1px solid transparent;
    display: flex;
    align-items: center;
    outline: 0;
    @each $breakpoint in $breakpoint-groups {
        $key: nth($breakpoint, 1);
        @include media-breakpoint-up($key) {
            @include calloutsize($calloutwidths, $key);
        }
    }

    p:nth-child(2) {
        margin-bottom: 15px;
    }
    &__icon {
        margin-right: 10px;
        display: inline-block;
    }
    &__wrapper {
        p:first-child {
            padding-bottom: 15px;
        }
    }
    p {
        margin: 0;
    }
    &__full {
        max-width: 100%;
    }
    &__error {
        @include common-theme(background-color, red, x-light);
        @include common-theme(border-color, red, dark);
    }
    &__success {
        @include common-theme(background-color, green, x-light);
        @include common-theme(border-color, green, dark);
    }
    &__txt-center {
        justify-content: center;
    }
    &__feature {
        @include common-theme(background-color, white);
        @include common-theme(border-color, blue, base);
        @include brand-text(bold);
        @include common-theme(color, blue, base);
        width: 100%;
        max-width: 100%;
        display: inline-flex;
        text-align: left;
        @include media-breakpoint-up(md1) {
            max-width: auto;
            width: auto;
        }
    }
    &__inline-error {
        @extend %type--jura;
        margin-top: 5px;
        margin-bottom: 0;
        max-width: 343px;
        @include media-breakpoint-up(md1) {
            max-width: 304px;
        }

        @include media-breakpoint-up(lg1) {
            max-width: 374px;
        }

        @include media-breakpoint-up(xl1) {
            max-width: 428px;
        }
    }
}

.callout--msgbottom {
    max-width: 640px;
}

.callout--product {
    @include common-theme(color, black);
    @include common-theme(border-color, black);
}

.callout--pointer {
    padding: 8px;
    margin-bottom: 0;
    position: relative;
    .callout__wrapper {
        p:first-child {
            padding-bottom: 0;
        }
    }
    &-left {
        margin-left: 20px;
        &:before {
            content: "";
            border-right: 10px solid;
            border-left: 10px solid transparent;
            border-bottom: 10px solid transparent;
            border-top: 10px solid transparent;
            @include common-theme(border-right-color, blue, x-light);
            position: absolute;
            left: -20px;
        }
    }
    &-right {
        margin-right: 20px;
        &:before {
            content: "";
            border-left: 10px solid;
            border-right: 10px solid transparent;
            border-bottom: 10px solid transparent;
            border-top: 10px solid transparent;
            @include common-theme(border-left-color, blue, x-light);
            position: absolute;
            right: -20px;
        }
    }
}