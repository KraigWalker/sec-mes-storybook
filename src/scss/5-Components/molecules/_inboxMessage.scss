.c-message {
    display: flex;
    margin-top: get-px(b);
    border: 1px solid bow-palette(grey, base, base);
    max-width: 640px;

    &.c-message--stacked {
        border-bottom: 0;
        margin-top: 0;
    }

    &:focus {
        @include on-focus();
    }
}

.c-message__icon {
    display: inline-flex;
    margin: get-px(a) 0 get-px(a) get-px(a);

    svg { @include general-theme(fill, blue, account); }
}

.c-message__summary {
    margin: get-px(a);
    width: calc(100% - 48px);
}

.c-message__summary.c-message__summary--draft {
    width: calc(100% - 16px);
}

.c-message__summary__head {
    display: flex;
    justify-content: space-between;
    margin-bottom: 0;
    position: relative;

    @include media-breakpoint-up(md1) {
        margin-bottom: get-px(a);
    }
}

.c-message__summary__head__title {
    display: flex;
    flex-flow: column;
    width: calc(100% - 68px);

    @include media-breakpoint-up(md1) {
        flex-flow: row;
    }
}

.c-message__summary__head__title.c-message__summary__head__title--draft {
    @include media-breakpoint-down(sm2) {
        width: calc(100% - 34px);
    }
}

.c-message__summary__head .c-message__summary__head__title__subject {
    @extend .type--uist;
    @include brand-heading(bold);
    
    margin: 0 get-px(a) get-px(a) 0;
    
    @include media-breakpoint-up(md1) {
        margin-bottom: 0;
    }

    @include media-breakpoint-down(sm2) {
        @include text-overflow();
        vertical-align: -webkit-baseline-middle;
        width: 100%;
    }

    &.c-message__summary__head__title__subject--read {
        @include brand-heading();
    }
}
.c-message__summary__head .c-message__summary__head__title__ref {
    display: block;
    // margin-bottom: get-px(a);
    @include media-breakpoint-up(md1) {
        display: inline;
        margin-bottom: 0;
    }
}

.c-message__summary__head .c-message__summary__head__actions {
    display: inline-flex;

    @include media-breakpoint-down(sm2) {
        align-items: flex-start;
    }

    &__reply,
    &__delete {
        @extend .type--bute;
        margin-left: get-px(a);
    }

    &__reply__txt,
    &__delete__txt {
        display: none;
        @include media-breakpoint-up(md1) {
            display: inline-block;
        }
    }

    &__reply__icon,
    &__delete__icon {
        display: inline-block;

        svg { @include general-theme(fill, blue, account); }
        
        @include media-breakpoint-up(md1) {
            display: none;
        }
    }
}

.c-message__summary {
    @extend .type--uist;

    &__account {
        @include text-overflow();
        // margin-bottom: get-px(a);
        width: 100%;
    } 

    &__date {
        margin-bottom: 0;
    }
}

.c-messagelist {
    padding: 0;
    max-width: 640px;
    
    &__wrapper {
        list-style: none;

        &:last-child {
            border-bottom: 1px solid bow-palette(grey, base, base);
        }
    }
}
