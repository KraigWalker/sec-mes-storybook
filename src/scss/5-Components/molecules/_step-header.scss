.centralised-container {
    max-width: 640px;
    margin: 0 auto;
    background-color: white;
}

.c-step-header {
    outline-width: 0;

    & :focus {
        outline-width: 0;
    }

     &__title {
        margin-top: 32px;
        margin-bottom: 16px;
        @extend %type--mull; 
        outline-width: 0;
        @include brand-heading(bold);
    }

    &__subtitle {
        margin-top: 16px;
        margin-bottom: 16px;
        font-weight: normal;
        @extend %type--orkney;
        
    }

    &__subtext {
        @extend %type--jura;
        margin-bottom: 24px;
        @include brand-text;
    }

    &__crumbs {
        margin-bottom: 16px; 
        margin-top: 16px;

            .c-step-header__link {
                @extend %type--jura; 
                @include brand-theme(color);
                @include brand-text;

            &:focus {
                @include on-focus();
            }

            .icon {
                vertical-align: middle;
            }
            .c-step-header__linkicon {
                margin-right: 5px;
                svg { @include brand-theme(fill);}
            }
            .back-link {
                text-decoration: underline;
            }
            .c-step-header__linktext {
                text-decoration: underline;
            }
        }

    }

    &__crumbs + &__title {
        margin-top: 16px;
    }

    &__controls {
        .c-step-header__control {
            @include general-theme(fill, blue, account);
            background: transparent;
            margin-bottom: 24px;

            &:hover,
            &:focus {
                background: transparent;
                text-decoration: none;
            }
            
            &:focus {
                @include general-theme(fill, blue, account);
            }

            &:active {
               @include general-theme(background-color, blue, account);
               @include common-theme(fill, white, base);
            }

            &.disabled {
                color: gray;
            }
        }

    }
}
.c-step-header--app-error {
    .c-step-title {
        display: flex;
        align-items: center;
        justify-content: center;
    h2 {
        margin: 40px 0 24px 10px;
        outline-width: 0;
        @include media-breakpoint-down(sm2){
            font-size: 20px;
            line-height: 28px;
        }
    }
    span {
        margin-bottom: -20px;
        svg {
            @include general-theme(fill, teal, base);
        }
    }
    }
    p {
        margin-bottom: 24px;
    }
}

//to fix skip links focus behind static header issue. 
#site-main-container{
	&::before{
		display: block;
		content: " ";
		margin-top: -240px;
		height: 240px;
		visibility: hidden;
	}
}