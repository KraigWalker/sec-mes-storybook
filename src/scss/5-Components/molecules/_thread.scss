
.c-thread {
    border: 1px solid bow-palette(grey, base, base);
    padding: get-px(b) get-px(b) get-px(b) get-px(c);
    margin-bottom: get-px(b);
    white-space: pre-wrap;
    max-width: 640px;

    .c-thread__body {
        margin: 0;
    }
}