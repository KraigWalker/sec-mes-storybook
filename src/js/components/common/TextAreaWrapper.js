import React from 'react';
import { Textarea } from 'web-ui-components/lib/atoms/forms';
import { ValidationMessage } from 'web-ui-components/lib/molecules/forms';
import { TextBody } from 'web-ui-components/lib/atoms/text';
import { TextStyled } from 'web-ui-components/lib/atoms/text';
import PropTypes from 'prop-types';

const calcCharsLeft = (maxChars, length) => maxChars - length;

class TextAreaWrapper extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    this.renderRemainingChar = this.renderRemainingChar.bind(this);

    const charsUsed = props.value ? props.value.length : 0;
    this.state = {
      charsLeft: calcCharsLeft(props.maxChars, charsUsed),
      message: props.value,
    };
  }
  render() {
    return (
      <div>
        <Textarea onChange={this.handleChange} rows={this.props.rows} cols={this.props.cols} id={this.props.id} maxLength="infinity" defaultValue={this.props.value} />
        {this.renderRemainingChar()}
      </div>
    );
  }

  renderRemainingChar() {
    const { charsLeft } = this.state;
    const { content, charsLeftDisplayThreshold } = this.props;

    if (charsLeft < 0) {
      return (
        <div>
          <TextBody>
            <ValidationMessage value={`${charsLeft} ${content.charLeft}`} hasIcon={false} />
          </TextBody>
          <TextBody>
            <ValidationMessage value={content.messageVal} />
          </TextBody>
        </div>
      );
    } else if (charsLeft <= charsLeftDisplayThreshold) {
      return (
        <TextStyled size="uist" className="u-padding-top-1">
          {charsLeft} {content.charLeft}
        </TextStyled>
      );
    }
  }

  handleChange(e) {
    const value = e.target.value;
    const charsLeft = calcCharsLeft(this.props.maxChars, value.length);

    this.setState({
      message: value,
      charsLeft,
    });

    this.props.onChange(e, charsLeft);
  }
}

TextAreaWrapper.propTypes = {
  onChange: PropTypes.func,
  maxChars: PropTypes.number.isRequired,
  content: PropTypes.object.isRequired,
  value: PropTypes.string,
  charsLeftDisplayThreshold: PropTypes.number.isRequired,
  rows: PropTypes.number.isRequired,
  cols: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
};

export default TextAreaWrapper;
