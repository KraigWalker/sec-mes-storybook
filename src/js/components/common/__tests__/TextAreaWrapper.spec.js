import { shallow } from 'enzyme';
import TextAreaWrapper from '../TextAreaWrapper';
import { TextStyled } from 'web-ui-components/lib/atoms/text';
import { ValidationMessage } from 'web-ui-components/lib/molecules/forms';

const basicProps = {
  id: 'TextArea',
  rows: 20,
  cols: 20,
  maxChars: 300,
  charsLeftDisplayThreshold: 100,
  onChange: jest.fn(),
  content: {},
};

describe('TextAreaWrapper snapshot', () => {
  let props = {
    ...basicProps,
    value: 'Some message value',
  };

  let wrapper = shallow(<TextAreaWrapper {...props} />);
  it('should match to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('TextAreaWrapper prop testing', () => {
  it('if passed value at which character count needs to display then show it', () => {
    let props = {
      ...basicProps,
      maxChars: 10,
      charsLeftDisplayThreshold: 5,
      value: 'hello u',
    };
    let wrapper = shallow(<TextAreaWrapper {...props} />);
    expect(wrapper.find(TextStyled).length).toBe(1);
  });

  it('if below character threshold dont show character count', () => {
    let props = {
      ...basicProps,
      maxChars: 10,
      charsLeftDisplayThreshold: 5,
      value: 'hell',
    };
    let wrapper = shallow(<TextAreaWrapper {...props} />);
    expect(wrapper.find(TextStyled).length).toBe(0);
  });

  it('less than 0 chars left then display error', () => {
    let props = {
      ...basicProps,
      maxChars: 10,
      charsLeftDisplayThreshold: 5,
      value: 'hello world!!!',
    };
    let wrapper = shallow(<TextAreaWrapper {...props} />);
    expect(wrapper.find(ValidationMessage).length).toBe(2);
  });

  it('has no error initially, then exceeds max chars', () => {
    let props = {
      ...basicProps,
      maxChars: 10,
      charsLeftDisplayThreshold: 5,
      value: '',
    };
    let event = {
      target: {
        value: 'Hello world!!!',
      },
    };
    let wrapper = shallow(<TextAreaWrapper {...props} />);
    wrapper.instance().handleChange(event);

    expect(wrapper.find(ValidationMessage).length).toBe(2);
  });

  it('has errors initially, then characters reduce to within threshold that char count displays', () => {
    let props = {
      ...basicProps,
      maxChars: 10,
      charsLeftDisplayThreshold: 5,
      value: 'Hello world!!!!',
    };
    let event = {
      target: {
        value: 'Hello w',
      },
    };
    let wrapper = shallow(<TextAreaWrapper {...props} />);
    wrapper.instance().handleChange(event);

    expect(wrapper.find(ValidationMessage).length).toBe(0);
    expect(wrapper.find(TextStyled).length).toBe(1);
  });
});
