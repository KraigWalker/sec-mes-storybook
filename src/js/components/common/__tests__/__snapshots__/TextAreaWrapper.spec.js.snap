// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaWrapper snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextAreaWrapper
    charsLeftDisplayThreshold={100}
    cols={20}
    content={Object {}}
    id="TextArea"
    maxChars={300}
    onChange={[MockFunction]}
    rows={20}
    value="Some message value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Textarea
          cols={20}
          defaultValue="Some message value"
          disabled={false}
          id="TextArea"
          maxLength="infinity"
          onChange={[Function]}
          rows={20}
        />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cols": 20,
          "defaultValue": "Some message value",
          "disabled": false,
          "id": "TextArea",
          "maxLength": "infinity",
          "onChange": [Function],
          "rows": 20,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Textarea
            cols={20}
            defaultValue="Some message value"
            disabled={false}
            id="TextArea"
            maxLength="infinity"
            onChange={[Function]}
            rows={20}
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cols": 20,
            "defaultValue": "Some message value",
            "disabled": false,
            "id": "TextArea",
            "maxLength": "infinity",
            "onChange": [Function],
            "rows": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
